Index: Python-2.7.2/Modules/Setup.dist
===================================================================
--- Python-2.7.2.orig/Modules/Setup.dist
+++ Python-2.7.2/Modules/Setup.dist
@@ -185,7 +185,7 @@ GLHACK=-Dclear=__GLclear
 #datetime datetimemodule.c	# date/time type
 #_bisect _bisectmodule.c	# Bisection algorithms
 
-#unicodedata unicodedata.c    # static Unicode character database
+unicodedata unicodedata.c    # static Unicode character database
 
 # access to ISO C locale support
 #_locale _localemodule.c  # -lintl
