Upstream-Status: Inappropriate [breakage in oe-core]
Signed-off-by: Ross Burton <ross.burton@intel.com>

--- a/configure.ac.orig	2012-10-02 14:57:27.736101398 +0100
+++ b/configure.ac	2012-10-02 14:57:37.416101335 +0100
@@ -946,7 +946,7 @@
 
 dnl === I18N ==================================================================
 
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.18])
 AM_GNU_GETTEXT([external])
 
 GETTEXT_PACKAGE="clutter-1.0"
