Upstream-Status: Pending

Index: pygtk-2.24.0/gtk/__init__.py
===================================================================
--- pygtk-2.24.0.orig/gtk/__init__.py	2011-04-01 11:37:25.000000000 +0100
+++ pygtk-2.24.0/gtk/__init__.py	2012-04-03 20:33:54.058918234 +0100
@@ -66,7 +66,8 @@
 
 keysyms = LazyModule('keysyms', locals())
 
-_init()
+if not hasattr(sys.modules['__main__'], 'python_launcher_enabled'):
+    _init()
 
 # CAPI
 _PyGtk_API = _gtk._PyGtk_API
